be/src/exec/pipeline/exchange/sink_buffer.h

```diff
@@ -40,58 +40,51 @@ class SinkBuffer {
             : _mem_tracker(state->instance_mem_tracker()),
               _brpc_timeout_ms(std::min(3600, state->query_options().query_timeout) * 1000),
               _num_remaining_sinkers(num_sinkers) {
-        _threads = ExecEnv::GetInstance()->pipeline_exchange_sink_thread_pool();
         for (const auto& dest : destinations) {
             const auto& dest_instance_id = dest.fragment_instance_id;
 
-            auto it = _num_sinkers_per_dest_instance.find(dest_instance_id.lo);
-            if (it != _num_sinkers_per_dest_instance.end()) {
+            auto it = _num_sinkers.find(dest_instance_id.lo);
+            if (it != _num_sinkers.end()) {
                 it->second += num_sinkers;
             } else {
-                _num_sinkers_per_dest_instance[dest_instance_id.lo] = num_sinkers;
+                _num_sinkers[dest_instance_id.lo] = num_sinkers;
 
                 // This dest_instance_id first occurs, so create other variable for this dest instance.
-                auto* closure = new CallBackClosure<PTransmitChunkResult, TUniqueId>();
-                closure->ref();
-                closure->addFailedHandler([this]() noexcept {
-                    {
-                        std::lock_guard<std::mutex> l(_mutex);
-                        _is_finishing = true;
-                    }
+                auto* closure = new CallBackClosure<PTransmitChunkResult, TUniqueId>(dest_instance_id);
```

Comment:
Seems you only use instance_id.lo, not instance_id.

Reply:
Yes, but a little meaningless if only pass `int64_t`